include/api/CDataFrameAnalyzer.h

```diff
@@ -69,7 +71,11 @@ class API_EXPORT CDataFrameAnalyzer {
     bool isControlMessage(const TStrVec& fieldValues) const;
     bool handleControlMessage(const TStrVec& fieldValues);
     void addRowToDataFrame(const TStrVec& fieldValues);
-    void writeResultsOf(const CDataFrameAnalysisRunner& analysis) const;
+    void reportProgress(const CDataFrameAnalysisRunner& analysis,
+                        core::CRapidJsonConcurrentLineWriter& writer) const;
+    void writeProgress(double progress, core::CRapidJsonConcurrentLineWriter& writer) const;
```

Comment:
nit: `writeProgress` and `reportProgress` seem very similar to me. How about `reportProgress` |-> `monitorProgress`.